client/packages/purchasing/src/purchase_order/DetailView/Tabs/GoodsReceived.tsx

```diff
@@ -1,67 +1,87 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import {
   ColumnAlign,
-  ColumnFormat,
-  DataTable,
   NothingHere,
-  useColumns,
   useNavigate,
   useParams,
   useTranslation,
   RouteBuilder,
+  ColumnDef,
+  ColumnType,
+  GoodsReceivedNodeStatus,
+  useNonPaginatedMaterialTable,
+  MaterialTable,
 } from '@openmsupply-client/common';
 import { AppRoute } from '@openmsupply-client/config';
 import {
   useGoodsReceivedList,
   GoodsReceivedRowFragment,
 } from '../../../goods_received/api';
+import { getGoodsReceivedStatusTranslator } from '../../../utils';
 
 export const GoodsReceived = () => {
   const t = useTranslation();
   const { purchaseOrderId } = useParams();
   const navigate = useNavigate();
 
   const {
-    query: { data, isError, isLoading },
+    query: { data, isFetching },
   } = useGoodsReceivedList({
     filterBy: { purchaseOrderId: { equalTo: purchaseOrderId } },
   });
 
-  const columns = useColumns<GoodsReceivedRowFragment>([
-    {
-      key: 'number',
-      label: 'label.number',
-      accessor: ({ rowData }) => rowData.number,
-      align: ColumnAlign.Right,
-    },
-    {
-      key: 'supplier',
-      label: 'label.supplier',
-      accessor: ({ rowData }) => rowData.supplier?.name,
-    },
-    {
-      key: 'status',
-      label: 'label.status',
-      accessor: ({ rowData }) => rowData.status,
-    },
-    {
-      key: 'supplierReference',
-      label: 'label.supplier-reference',
-      accessor: ({ rowData }) => rowData.supplierReference,
-    },
-    {
-      key: 'createdDateTime',
-      label: 'label.created-datetime',
-      format: ColumnFormat.Date,
-      accessor: ({ rowData }) => rowData.createdDatetime,
-    },
-    {
-      key: 'receivedDateTime',
-      label: 'label.received-date',
-      format: ColumnFormat.Date,
-      accessor: ({ rowData }) => rowData.receivedDatetime,
-    },
-  ]);
+  const columns = useMemo(
+    (): ColumnDef<GoodsReceivedRowFragment>[] => [
+      {
+        header: t('label.number'),
+        accessorKey: 'number',
+        columnType: ColumnType.Number,
+        size: 100,
+        enableSorting: true,
+        enableColumnFilter: true,
+      },
+      {
+        header: t('label.supplier'),
+        accessorKey: 'supplier.name',
+        enableSorting: false,
+      },
+      {
+        header: t('label.status'),
+        id: 'status',
+        size: 120,
+        accessorFn: row => getGoodsReceivedStatusTranslator(t)(row.status),
+        filterVariant: 'select',
+        filterSelectOptions: Object.values(GoodsReceivedNodeStatus).map(
+          status => ({
+            value: status,
+            label: getGoodsReceivedStatusTranslator(t)(status),
+          })
+        ),
+        enableSorting: true,
+        enableColumnFilter: true,
+      },
+      {
+        header: t('label.supplier-reference'),
+        accessorKey: 'supplierReference',
+        align: ColumnAlign.Left,
```

Comment:
This type not used anymore (it's just string literals I think), and you don't need it for "left" as that's the default